### PR TITLE
Document site contribution guidelines

### PR DESCRIPTION
**Pull request by SDBrett**
Updates: #1891 

Documentation for website contribution. 

Looking for initial feedback 

Signed-off-by: Brett Johnson <brett@sdbrett.com>

### CODE REVIEW

File: SITE_CONTRIBUTION.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        ├── objects
+        ├── settings
+        └── utilities
+```
+
+## Link Insertion Standards
+
+Usage of links within a page has been standardized to simplify the process of finding and fixing links as required.
+
+Avoid using inline links, instead, use a reference table the bottom of the file; making it easier to find and update links. [Reference Links][3]. There are some exceptions to this rule, such as depreciation warnings.
```
Comment (youngnick) [inline SITE_CONTRIBUTION.md]: Nit: Sadly, this should be `deprecation warnings`, not `depreciation warnings`. Thanks, English, for having such a large difference in meaning by one letter.

File: SITE_CONTRIBUTION.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+A full list of variables can be found in the [jekyll-github-metadata documentation][4].
+
+**Linking to an external page:**
+
+Links to external pages are done using the full URL, `[1]: https://projectcontour.io` for example.
+
+## Proofread and spell check
+
+If you're making any content submissions, please proofread and spell check your work.
```
Comment (youngnick) [inline SITE_CONTRIBUTION.md]: Maybe note here that running `make check` will spell check for you?

Comment (SDBrett) [reply SITE_CONTRIBUTION.md]: I'll add a note, though I would like people to do their own checking as well

File: SITE_CONTRIBUTION.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+You can use a local instance of Jekyll to build and test the site. This section assumes that you already have Ruby 2.6 or higher installed.
+
+```
+$ cd site
+$ bundle install
+$ bundle exec jekyll --livereload
+```
+
+You might find that your Gemfile.lock has changed after running bundle install, mainly if your version of bundler is higher than 1.7.3. Discard any unintended changes to the file.
+
```
[the remark this answers is not in the document]
Comment (SDBrett) [reply SITE_CONTRIBUTION.md]: Done

File: SITE_CONTRIBUTION.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+On Ubuntu you will need these packages:
+
+- ruby
+- ruby-dev
+- ruby-bundler
+- build-essential
+- zlib1g-dev
+- nginx (or apache2)
+
+Install [Jekyll](https://jekyllrb.com) and plug-ins in one fell
```
Comment (jpeach) [inline SITE_CONTRIBUTION.md]: nit: it seems inconsistent that this doesn't use footnote style links

File: SITE_CONTRIBUTION.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+1. Clone down your own fork, or clone the main repo `git clone https://github.com/projectcontour/contour` and add your own remote.
+2. `cd site`
+3. `rbenv local 2.6.3`
+4. `bundle install`
+5. Serve the site and watch for markup/sass changes `jekyll serve --livereload`. You may need to run `bundle exec jekyll serve --livereload`.
+6. View your website at http://127.0.0.1:4000/
+
+You might find that your Gemfile.lock has changed after running bundle install, mainly if your version of bundler is higher than 1.7.3. Discard any unintended changes to the file.
+
+*IMPORTANT:* The bundler version specified in `site/Gemfile.lock` MUST be 1.7.3. Changing to a newer version causing issues with the Netlify build process.
```
[the remark this answers is not in the document]
Comment (SDBrett) [reply SITE_CONTRIBUTION.md]: Looks like I can remove that section as I have builds working with 2.0.2 on Netlify.

Will resolve

File: SITE_CONTRIBUTION.md (hunk 4 above)
Comment (SDBrett) [reply SITE_CONTRIBUTION.md]: Good pickup

**Pull request merged**